python/mlx/nn/layers/linear.py

```diff
@@ -6,6 +6,21 @@
 from mlx.nn.layers.base import Module
 
 
+class Identity(Module):
```

Comment:
Could you explain in which cases this will be used? And maybe provide an example?

Reply:
Here are some use cases for the identity layer:
1. Removing parts of existing neural networks:
- _Example_: Directly obtain features instead of class logits from pretrained network.
```python
model = models.resnet18()
model.fc = nn.Identity() # Replace final linear layer with Identity
features = model(input_tensor)
```
2. Conditional Computation:
- Enables different computations based on conditions while keeping model structure.
- _Example_: batchnorm as condition
```python
batch_norm = nn.BatchNorm2d
if dont_use_batch_norm:
    batch_norm = Identity

nn.Sequential(
    ...
    batch_norm(N, momentum=0.05),
    ...
)
```
References:
- https://github.com/pytorch/pytorch/issues/9160#issuecomment-456766137
- https://github.com/pytorch/pytorch/pull/19249
- https://discuss.pytorch.org/t/what-is-nn-identity-used-for/125457/14

Reply:
Makes sense, thanks for the input. I like the first use case, that's clever.